AK/Time.cpp

```diff
@@ -38,6 +41,299 @@ unsigned day_of_week(int year, unsigned month, int day)
     return (year + year / 4 - year / 100 + year / 400 + seek_table[month - 1] + day) % 7;
 }
 
+#ifndef KERNEL
+Optional<struct tm> convert_formatted_string_to_timespec(StringView input, StringView format, size_t& string_pos)
+{
+    [[maybe_unused]] size_t format_pos = 0;
+    auto tm_or_void = convert_formatted_string_to_timespec(input, format, string_pos, format_pos);
+    if (!tm_or_void.has_value())
+        return {};
+    return tm_or_void.release_value();
```

Comment:
```suggestion
    return tm_or_void;
```

Could also be a return without storing the value in a variable first.